File: CPP_Programming/Day_7/program59.cpp
// Description : Write a program that accepts array size , elements from the user ,
//               & returns the summation of the odd elements from the array.

#include<iostream>
using namespace std;

class ArrayX
{
    public :
        int * Arr;
        int iSize;

    ArrayX(int * Arr , int iSize)
    {
        this->Arr = Arr;
        this->iSize = iSize;
    }

    int SumOdd()
    {
        int iCnt = 0;
        int iSumOdd = 0;

        for(iCnt = 0 ; iCnt < iSize ; iCnt++)
        {
            if(Arr[iCnt] % 2 != 0)
            {
                iSumOdd = iSumOdd + Arr[iCnt];
            }
        }

        return iSumOdd;

    }

    void Accept()
    {
        int iCnt = 0;

        cout<<"Enter the elements : "<<"\n";
        for(iCnt = 0 ; iCnt < iSize ; iCnt++)
        {
            scanf("%d",&Arr[iCnt]);
        }
    }

     void Display()
    {
        int iCnt = 0;

        cout<<"Elements from the array are : "<<"\n";
        for(iCnt = 0 ; iCnt < iSize ; iCnt++)
        {
            printf("%d\t",Arr[iCnt]);
        }
        cout<<"\n";
    }
};


int main()
{
    int * Arr = NULL;
    int iLength = 0;
    int iRet = 0;

    cout<<"Enter the size of the array : ";
    cin>>iLength;

    Arr = new int[iLength];

    ArrayX aobj(Arr , iLength);

    aobj.Accept();
    aobj.Display();

    iRet = aobj.SumOdd();
    cout<<"Summation of odd elements is : "<<iRet<<"\n";

    return 0;
}

File: Java_Programming/Day_7/program59.java
// Description : Write a program that accepts array size , elements from the user ,
//               & returns the summation of the odd elements from the array.


import java.util.*;

class ArrayX
{
    public int Arr[];
    public int iSize = 0;
    Scanner sobj = new Scanner(System.in);

    ArrayX(int iSize)
    {
        this.iSize = iSize;

        Arr = new int[iSize];
    }

    int SumOdd()
    {
        int iCnt = 0;
        int iSumOdd = 0;

        for(iCnt = 0 ; iCnt < iSize ; iCnt++)
        {
            if(Arr[iCnt] % 2 != 0)
            {
                iSumOdd = iSumOdd + Arr[iCnt];
            }
        }

        return iSumOdd;
    }

    void Accept()
    {
        int iCnt = 0;
        System.out.println("Enter the elements : ") ;

        for(iCnt = 0 ; iCnt < iSize ; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }
    }

    void Display()
    {
        int iCnt = 0;

        System.out.println("Elements are : ");
        for(iCnt = 0 ; iCnt < iSize ; iCnt++)
        {
            System.out.print(Arr[iCnt] + "\t");
        }
        System.out.println();
    }
}

class program59
{
    public static void main(String A[])
    {
        int iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the array size : ");
        int iLength = sobj.nextInt();

        ArrayX aobj = new ArrayX(iLength);

        aobj.Accept();
        aobj.Display();

        iRet = aobj.SumOdd();
        System.out.println("Summation of odd elements is : "+iRet);

        sobj.close();
    }
}